indexer/src/utils/filter.test.ts

```diff
@@ -71,3 +72,94 @@ Deno.test("ethValidationFailed: incorrect data length", () => {
   const failed = ethValidationFailed(x);
   assertFalse(failed);
 });
+
+Deno.test("isKakarotTransaction: no calldata", () => {
+  const transaction: Transaction = {
+    invokeV1: {
+      senderAddress: "0x01",
+      calldata: []
+    },
+    meta: {
+      hash: "0x01",
+      maxFee: "0x01",
+      nonce: "0x01",
+      signature: ["0x1", "0x2", "0x3", "0x4", "0x32"],
+      version: "1",
+    },
+  };
+  const failed = isKakarotTransaction(transaction);
+  assertFalse(failed);
+});
+
+Deno.test("isKakarotTransaction: no `to` field in calldata", () => {
+  const starknetTxCalldata: `0x${string}`[] = [
+    "0x1"
+  ];
+  const transaction: Transaction = {
+    invokeV1: {
+      senderAddress: "0x01",
+      calldata: starknetTxCalldata
+    },
+    meta: {
+      hash: "0x01",
+      maxFee: "0x01",
+      nonce: "0x01",
+      signature: ["0x1", "0x2", "0x3", "0x4", "0x32"],
+      version: "1",
+    },
+  };
+  const failed = isKakarotTransaction(transaction);
+  assertFalse(failed);
+});
+
+Deno.test("isKakarotTransaction: `to` address not matching KAKAROT_ADDRESS", () => {
+  const serializedTx: `0x${string}`[] = [];
+  const starknetTxCalldata: `0x${string}`[] = [
+    "0x2",
+    ...serializedTx,
+  ];
```

Comment:
you're missing a `to` value here. `starknetTxCalldata` will just be `[0x2]` while the `to` should be at index 1.